src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php

```diff
@@ -103,16 +104,29 @@ protected function processItem($WpObject)
         }
 
         $this->debug('Added guest information', $callData);
+
+        $ShopModule = $this->storekeeper_api->getModule('ShopModule');
+
         /*
          * Order products
          */
         if (!$isUpdate) {
             $callData['order_items'] = $this->getOrderItems($WpObject);
             $this->debug('Added order_items information', $callData);
         } else {
-            $callData['order_items__do_not_change'] = true;
-            $callData['order_items__remove'] = null;
-            $this->debug('Update, ignored order_items', $callData);
+            $storekeeperId = $this->get_storekeeper_id();
+            $storekeeperOrder = $ShopModule->getOrder($storekeeperId, null);
+
+            $hasDiscrepancy = $this->checkOrderDiscrepancy($WpObject, $storekeeperOrder);
+            // Only update order items if they have discrepancy and order is not paid yet
+            if ($hasDiscrepancy && !$storekeeperOrder['is_paid']) {
+                $callData['order_items'] = $this->getOrderItems($WpObject);
+                $this->debug('Updated order_items information due to discrepancy with the backoffice order items', $callData);
+            } else {
```

Comment:
what if it's paid and has differences? maybe the task should fail than, cos it didn't sync

Reply:
Added new `elseif` condition to throw an exception if order has discrepancy but is already paid.